<launch>
  <arg name="sim_name" default="gazebo"/>
  <arg name="map_name" default="campus"/>
  <arg name="config" default="indoor"/>
  <arg name="stateEstimationTopic" default="/mavros/local_position/odom"/>
  <arg name="depthCloudTopic" default="/registered_scan"/>
  <arg name="depthCloudDelay" default="0"/>
  <arg name="depthCamPitchOffset" default="0"/>
  <arg name="depthCamXOffset" default="0"/>
  <arg name="depthCamYOffset" default="0"/>
  <arg name="depthCamZOffset" default="0"/>
  <arg name="trackingCamBackward" default="false"/>
  <arg name="trackingCamXOffset" default="0"/>
  <arg name="trackingCamYOffset" default="0"/>
  <arg name="trackingCamZOffset" default="0"/>
  <arg name="trackingCamScale" default="1.0"/>
  <arg name="vehicleX" default="0"/>
  <arg name="vehicleY" default="0"/>
  <arg name="vehicleZ" default="1.5"/>
  <arg name="vehicleYaw" default="0"/>
  <arg name="use_gazebo" default="true"/>
  <arg name="gazebo_gui" default="false"/>
  <arg name="realtime_factor" default="1.0"/>
  <arg name="checkTerrainConn" default="false"/>

  <include file="$(find ps3joy)/launch/ps3.launch" />

  <include file="$(find local_planner)/launch/local_planner_$(arg config).launch" >
    <arg name="stateEstimationTopic" value="$(arg stateEstimationTopic)" />
    <arg name="depthCloudTopic" value="$(arg depthCloudTopic)" />
    <arg name="depthCloudDelay" value="$(arg depthCloudDelay)" />
    <arg name="depthCamPitchOffset" value="$(arg depthCamPitchOffset)" />
    <arg name="depthCamXOffset" value="$(arg depthCamXOffset)" />
    <arg name="depthCamYOffset" value="$(arg depthCamYOffset)" />
    <arg name="depthCamZOffset" value="$(arg depthCamZOffset)" />
    <arg name="trackingCamBackward" value="$(arg trackingCamBackward)" />
    <arg name="trackingCamXOffset" value="$(arg trackingCamXOffset)" />
    <arg name="trackingCamYOffset" value="$(arg trackingCamYOffset)" />
    <arg name="trackingCamZOffset" value="$(arg trackingCamZOffset)" />
    <arg name="trackingCamScale" value="$(arg trackingCamScale)" />
    <arg name="sim_name" value="$(arg sim_name)"/>
  </include>

  <include file="$(find terrain_analysis)/launch/terrain_analysis.launch" />

  <include file="$(find terrain_analysis_ext)/launch/terrain_analysis_ext.launch" >
    <arg name="checkTerrainConn" value="$(arg checkTerrainConn)"/>
  </include>

  <include file="$(find vehicle_simulator)/launch/vehicle_simulator_gazebo.launch" >
    <arg name="sensorPitch" value="$(arg depthCamPitchOffset)" />
    <arg name="vehicleX" value="$(arg vehicleX)" />
    <arg name="vehicleY" value="$(arg vehicleY)" />
    <arg name="vehicleZ" value="$(arg vehicleZ)" />
    <arg name="vehicleYaw" value="$(arg vehicleYaw)" />
    <arg name="use_gazebo" value="$(arg use_gazebo)" />
    <arg name="gui" value="$(arg gazebo_gui)" />
    <arg name="map_name" value="$(arg map_name)" />
    <arg name="realtime_factor" value="$(arg realtime_factor)" />
    <arg name="sim_name" value="$(arg sim_name)"/>
  </include>

  <include file="$(find visualization_tools)/launch/visualization_tools_gazebo.launch" >
    <arg name="map_name" value="$(arg map_name)"/>
  </include>

  <!--node launch-prefix="nice" pkg="rviz" type="rviz" name="rvizAA" args="-d $(find vehicle_simulator)/rviz/vehicle_simulator_gazebo.rviz" respawn="true"/-->

  <node pkg="tf" type="static_transform_publisher" name="odomTransPublisher" args="0 0 0 0 0 0 map odom 10"/>

</launch>
